### PR TITLE
Make simulations work on Vivado xsim. Tested on Vivado 2017.1

### PR DESCRIPTION
**Pull request by xalogic-linus**
Created a new directory sim_vivado in order not to mess up with the existing VCS environment.
TB files and RTL needed some ifdefs to be compatible with Vivado XSIM.
- $value$plugargs compatibility
- $sformatf not supported by Vivado

TODO:
- about 100+ RTL files in VLIB requires more ifdefs for $value$plusargs. This seems to not affect basic simulations. I will work on these next.

### CODE REVIEW

**Comment by jwise** [comment]
This looks great.  Let me check with our legal team if we need you to sign a contributor license agreement to merge this.  Is this done on behalf of XaLogic, or was this done entirely in your own personal capacity?  If the former, can you confirm that you have permission to license this under the Open NVDLA License, and that you do in fact do so?

**Comment by xalogic-linus** [comment]
I would like to address this privately through email. Can you provide me an email I should send my response to?

**Comment by jwise** [comment]
Yes, you can send us mail at nvdla@nvidia.com.

**Comment by jwise** [comment]
I haven't forgotten about this.  We're working on a branch into the 'configurable' branch, which will diverge from the 'dlav1' branch; I intend to integrate this into the dlav1 branch once I've got that forked off and set up properly.  Thanks for your patience!

**Comment by xalogic-linus** [comment]
Hi Joshua,

No problem. Thanks and appreciate your update.

Linus

Sent from my Mobile

-------- Original Message --------
Subject: Re: [nvdla/hw] Make simulations work on Vivado xsim. Tested on Vivado 2017.1 (#49)
From: Joshua Wise
To: nvdla/hw
CC: Linus Kerk ,Author


I haven't forgotten about this. We're working on a branch into the 'configurable' branch, which will diverge from the 'dlav1' branch; I intend to integrate this into the dlav1 branch once I've got that forked off and set up properly. Thanks for your patience!

File: vmod/vlibs/RANDFUNC.vlib [hunk 1]
```diff
@@ -129,14 +138,21 @@ input in1;
...
+`endif
 begin
 	//	hash = 32'h9e3779b9;
 	hash = 0;
 	// Convert current module hierarchical name to a reg.
 	// Note $sformatf() converts strings right to left.  i.e.
 	// Return value contains data from right most data of the string.
+	`ifdef VIVADO
+	s = $sformatf ("%m");
+	`else
```
Comment (jwise) [inline vmod/vlibs/RANDFUNC.vlib]: This seems not quite right (lines 152 and 153 transposed?)

Comment (xalogic-linus) [reply vmod/vlibs/RANDFUNC.vlib]: You are right, line 152 and 153 should be swopped. I had made the fix in my branch.

Sorry about that.

Comment (jwise) [reply vmod/vlibs/RANDFUNC.vlib]: Ok, squashed this commit locally.  Working on batching this up with a handful more changes.

**Comment by jwise** [comment]
Ok, I have pulled this into the 'nvdlav1' branch, which is the new branch for the "non-configurable" NVDLA release that we're working on.  Thanks!

**Comment by jwise** [comment]
FYI -- I don't have an Xilinx license to test this in a backport to the 'configurable' branch (coming in the next week or two), so this change isn't going to make it into the first version of that.  (Additionally, the testbench has been substantially revamped since then.)  But, I think that there should soon be some Xilinx synthesis compatibility changes in preparation to run on Amazon EC2.

**Pull request closed without merging**